### PR TITLE
feat(@covalent/coding-standards): create a @covalent/coding-standards package

### PR DESCRIPTION
**Pull request by christianmemije**
## Description

- Creates a `@covalent/coding-standards` package that we can publish and re-use in other applications in order to pin the configs to packages and package versions.
- Makes husky config re-usable
- Removed rules of stylelint and tslint configs that conflicted with prettier to make extending the configs simpler.

#### Test Steps

- [ ] `npm ci`
- [ ] `npm run build:lib`
- [ ] ensure all files are under `deploy/platform/coding-standards`
- [ ] cd into that dir and do `npm pack`
- [ ] go into another project and do npm install `<path to the tgz file created in command above>`

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

### CODE REVIEW

File: src/platform/coding-standards/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "@covalent/coding-standards",
+  "version": "0.0.0-COVALENT",
+  "description": "Teradata Covalent shared configurations",
```
Comment (emoralesb05) [inline src/platform/coding-standards/package.json]: Lets use a better description for this.

File: package.json [hunk 2]
```diff
@@ -40,12 +40,6 @@
     "prettier:check": "./node_modules/.bin/prettier --check './**/*.{ts,js,json,css,scss,yml,html}'",
     "commitlint": "node ./node_modules/@commitlint/travis-cli/lib/cli.js"
   },
-  "husky": {
```
Comment (emoralesb05) [inline package.json]: We dont need this?

File: package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "lint": "npm run tslint && npm run stylelint",
```
Comment (emoralesb05) [inline package.json]: Should this run prettier check?

File: package.json [hunk 4]
```diff
@@ -14,7 +14,7 @@
...
     "tslint": "./node_modules/.bin/tslint -p ./tsconfig.json --format codeFrame -c ./tslint.json './src/**/*.ts'",
     "stylelint": "./node_modules/.bin/stylelint 'src/**/*.scss' '!**/_layout.scss' '!**/markdown.component.scss' --config stylelint.config.js --syntax scss",
-    "preinstall": "rm -rf node_modules tmp deploy dist",
+    "preinstall": "rm -rf tmp deploy dist",
```
Comment (emoralesb05) [inline package.json]: We can probably remove this hook.

File: package.json (hunk 2 above)
Comment (christianmemije) [reply package.json]: We added a `.huskyrc.js` file where we can extend the default husky config inside coding-standards

File: package.json (hunk 3 above)
Comment (christianmemije) [reply package.json]: Probably, but I would say that is slightly out of scope of this PR

File: package.json (hunk 4 above)
Comment (christianmemije) [reply package.json]: Removed

File: src/platform/coding-standards/package.json (hunk 1 above)
Comment (christianmemije) [reply src/platform/coding-standards/package.json]: Updated

**Pull request merged**